### ISSUE TITLE
Missing ComponentType in MethodInfo.getParameterTypeStrs()

### ISSUE THREAD

**Issue by dantesun** [lukehutch/fast-classpath-scanner#152]
```
public void method(@ParamAnnotation0 final List<String> annotatedValue0,
            @ParamAnnotation1 @ParamAnnotation2 final String annotatedValue1) {
    }
```
MethodInfo.getParameterTypeStrs() simply returns `[List, String]`. But `annotatedValue0` is a `List<String>`.

This problem seems also exists in other types returned like `FieldInfo#getTypeStr` or `MethodInfo.getReturnTypeStr` ...

**Comment by lukehutch** [comment]
Parameter types are much harder to obtain from the classfile (the type parameters are stored in a different place from the types), and they are often not even present in the classfile (due to type erasure, or in earlier versions of the VM). But let me see what can be done for cases where the info does exist.

**Comment by dantesun** [comment]
I am not an expert on JVM class format. I am not sure if you need this or not. But here is my two cents.
https://stackoverflow.com/questions/937933/where-are-generic-types-stored-in-java-class-files

**Comment by lukehutch** [comment]
Yes, correct, it's in the Signature attribute. But it's a bit more complicated for method parameters, I think. I'll look into it.

**Comment by lukehutch** [comment]
Fixed and released in version 2.8.2.
